### ISSUE TITLE
Newlines before media queries not kept

### ISSUE THREAD

**Issue by codedbypaul** [Glavin001/atom-beautify#1103]
# Description

When a @media query follows either a {} rule or another @media the new line is removed. This causes scss-lint to complain.

I have 'Preserve newlines' ticked in the config options and it works with {} rules.

# Expected Results

The beautified code should have looked like:

```
.foo {
  color: green;
}

@media all and (min-width: 12em) {
  .foo {
    color: red;
  }
}

@media all and (min-width: 48em) {
  .foo {
    color: blue;
  }
}
```

It ends up looking like:
```
.foo {
  color: green;
}
@media all and (min-width: 12em) {
  .foo {
    color: red;
  }
}
@media all and (min-width: 48em) {
  .foo {
    color: blue;
  }
}
```

# Steps to Reproduce

1. Add code to Atom editor
2. Run command `Atom Beautify: Beautify Editor`
3. This beautified code does not look right!

# Debug

Here is a link to the `debug.md` Gist: https://gist.github.com/anonymous/656ccca2e5e338c48ca7e63b0b4a015b

# Checklist

- [x] I have tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [x] I have reloaded (or restarted) Atom to ensure it is not a caching issue
- [x] Searched for existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues
so I know this is not a duplicate issue
- [x] Generated debugging information and added link for `debug.md` Gist to this issue

**Comment by prettydiff** [comment]
I looked at the debug.md and discovered:

* Selected beautifier is Pretty Diff
* Language is SCSS
* Desired option `preserve_newlines` is `true`

This is likely a defect in Pretty Diff or Atom Beautify.  I am going to troubleshoot.

**Comment by prettydiff** [comment]
I can reproduce the defect in the latest version of Pretty Diff.

**Comment by prettydiff** [comment]
The resolution is on line: https://github.com/prettydiff/prettydiff/blob/master/lib/csspretty.js#L1852

This line:

    if (lines[a] > 1 && options.compressedcss === false && (types[a] === "start" || types[a] === "end" || types[a] === "selector" || types[a] === "comment" || types[a] === "property" || types[a].indexOf("external") > -1)) {
                    if (options.cssinsertlines === true && types[a] === "selector" && types[a - 1] !== "comment") {

Should be:

    if (lines[a] > 1 && options.compressedcss === false && (types[a] === "start" || types[a] === "end" || types[a] === "selector" || types[a] === "comment" || types[a] === "property" || types[a] === "propvar" || types[a].indexOf("external") > -1)) {

The difference being the inclusion of internal type `propvar` being included for evaluation in the area of the beautifier that handles empty lines.  This change is pushed to branch 2.1.1 and will be merged to master soon.

**Comment by prettydiff** [comment]
The resolution is published to Pretty Diff release 2.1.1.  It will not be automatically available to Atom Beautify until I can publish https://github.com/prettydiff/biddle

**Comment by prettydiff** [comment]
Keeping everybody informed I have fixed biddle.  Check it out now - https://github.com/prettydiff/biddle

I will republish Pretty Diff using biddle in production tomorrow.  I am not ready to give it to Atom Beautify just yet though.  I actually need users to play around with this thing for a bit so that I can get a sense of where the pain points are and just how bad it works for fails for other people.  I wrote a getting started document at https://github.com/prettydiff/biddle/blob/master/documentation/gettingstarted.md

The documentation might be easier to read from the command line though as I included a markdown to CLI parser in biddle.  The number at the end defines the word wrap limit:

    node biddle markdown documentation/gettingstarted.md 80
